restlicsharpclient/restliclienttest/RequestUrlBuilderTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿/*
+   Copyright (c) 2017 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+using System;
+using System.Collections.Generic;
+
+using restlicsharpclient.restliclient.util.url;
+using restlicsharpclient.restliclient.request;
+using restlicsharpclient.restliclient.request.builder;
+using restlicsharpclient.restliclient.response;
+using com.linkedin.restli.test.api;
+
+namespace restlicsharpclient.restliclienttest
+{
+    [TestClass]
+    public class RequestUrlBuilderTests
+    {
+        const string complexString = " () :', hello123%=&+?/";
+        const string complexStringEscapedQuery = "%20%28%29%20%3A%27%2C%20hello123%25%3D%26%2B?/";
+        const string complexStringEscapedPath = "%20%28%29%20%3A%27%2C%20hello123%25=&+%3F%2F";
+
+        [TestMethod]
+        public void BuildComplexUrl()
```

Comment:
Add a test for CREATE methods too. Will add one for FINDER in #11 once this PR is merged